internal/tracker/tracker.go

```diff
@@ -154,11 +154,52 @@ func Track(ctx context.Context, urls []string, input []byte, prune bool, freshen
 	}
 	t.TaskBundles = map[string][]bundleRecord{}
 
+	imageUrls, gitUrls := groupUrls(urls)
+
+	if err := trackImageReferences(ctx, imageUrls, &t, freshen); err != nil {
+		return nil, err
+	}
+
+	if err := trackGitReferences(ctx, gitUrls, &t); err != nil {
+		return nil, err
+	}
+
+	t.filterBundles(prune)
+
+	t.setExpiration()
+
+	if err := t.convertToOldFormat(); err != nil {
+		return nil, err
+	}
+
+	return t.Output()
+}
+
+func groupUrls(urls []string) ([]string, []string) {
+	imgs := make([]string, 0, len(urls))
+	gits := make([]string, 0, len(urls))
+	for _, u := range urls {
+		if strings.HasPrefix(u, "git+") {
+			gits = append(gits, u)
+		} else {
+			imgs = append(imgs, u)
+		}
+	}
+
+	return imgs, gits
+}
+
+func trackImageReferences(ctx context.Context, urls []string, t *Tracker, freshen bool) error {
```

Comment:
NIT: Why not make this a method on the `Tracker` type? (Same for `trackGitReferences`)